plugins/sign-on/openid-connect-bcc-customizations/bcc-signon.php

```diff
@@ -78,11 +70,21 @@ private function load_dependencies() {
 	 * Create the menu item
 	 */
 	public function bcc_signon_plugin_create_menu() {
+		add_action ( 'admin_init', function() {
```

Comment:
Does this allow anyone who adds ?delete_subscribers=true to the querystring to delete all subscribers? 
I think admin_init gets run whenever you try to load an admin screen... which doesn't necessarily mean you are an admin.

Reply:
See comment below

Reply:
No. From [here](https://wordpress.org/support/article/roles-and-capabilities/#manage_options) you can see that only the super admin and the administrator can go to WordPress Settings page (options-general.php) i.e. have the manage_options capability. I've also tested that and the other roles will get a 403 (Sorry, you are not allowed to access this page).

Reply:
Just depends if this script is run before or after the 403  response is generated...

Reply:
As I see from options-general.php, there is a check for the manage_options capability in the first lines of the file, and, as I said, I also tested it and got 403 for a non super admin or non administrator when I accessed the page, but if you think it makes sense, I can add a check for admin rights in our code.

Reply:
Could add a check to be on the safe side. It depends if the admin_init hook is run before or after checking user permissions. Theoretically, you could delete all the subscribers and then get a 403 after that

Reply:
The point I was trying to make is that this would not happen since I understand that the check of administrator rights is before admin_init. Also, the risk is not so important, I think.
Anyway, I can add the check for admin rights.